file.carp

```diff
@@ -258,6 +258,27 @@ writable.")
         (Result.Success 0))
       (Result.Error (fmt "The file “%s” is not writable" (name f)))))
 
+  (doc read 
+    "Reads len values from a file."
+    ""
+    ("This function is generic in its return argument. You can " false)
+    ("provide a custom reader implementation by implementing the " false)
+    "read-from-file interface."
+    ""
+    ("The surrounding context typically determine's the return type." false)
+    ("if no implementation of read-from-file exists for the type, you'll " false)
+    "need to provide one. You can also use `the` to specify the return type."
+    "For example (to read the first 3 bytes of the file):"
+    "```"
+    "(match (File.open-with \"test-file.txt\" \"r\")"
```

Comment:
To avoid having to write it twice, you can put the `the` in the first clause of `match`

Reply:
Thanks!